### PR TITLE
webgl: vertex array, buffer and texture cleanup

### PR DESCRIPTION
**Pull request by hatal175**
As @adrian17 and others discussed in discord, webgl leaks vertex array and buffers on replace_shape. I think his pull request should deal with that and hopefully make us see less UnableToCreateVAO crashes.

### CODE REVIEW

**Comment by hatal175** [comment]
Should I add deleting the bitmap texture to this commit as well?

**Review by Aaron1011 — APPROVED** [review]

File: render/webgl/src/lib.rs [hunk 1]
```diff
@@ -988,7 +1005,9 @@ impl RenderBackend for WebGlRenderBackend {
     }
 
     fn unregister_bitmap(&mut self, bitmap: BitmapHandle) {
-        self.bitmap_registry.remove(&bitmap);
+        if let Some(entry) = self.bitmap_registry.remove(&bitmap) {
```
Comment (relrelb) [inline render/webgl/src/lib.rs:1008]: I think we can `impl Drop for RegistryData`, and move `delete_texture` to there.
This is both more clear and less error-prone.

File: render/webgl/src/lib.rs [hunk 2]
```diff
@@ -685,6 +685,22 @@ impl WebGlRenderBackend {
         };
     }
 
+    fn delete_mesh(&self, mesh: &Mesh) {
```
Comment (relrelb) [inline render/webgl/src/lib.rs:688]: Similarly, this can be moved to `impl Drop for Draw`.

**Comment by relrelb** [comment]
We can leverage Rust's `drop` by implementing wrappers for `WebGlBuffer` and `WebGlTexture`. For example:
```rs
struct OwnedWebGlBuffer(WebGlBuffer, &WebGlRenderBackend);

impl Drop for OwnedWebGlBuffer {
    fn drop(&mut self) {
        self.1.gl.delete_buffer(Some(&self.0));
    }
}
```
This is should be cleaner and less error-prone.

**Comment by hatal175** [comment]
I;m not sure about this. The code as you wrote would require me to clone the gl instance. What are the consequences of that?

**Comment by relrelb** [comment]
> I;m not sure about this. The code as you wrote would require me to clone the gl instance. What are the consequences of that?

It should be very cheap - `wasm_bindgen::JsValue` is merely an index to a "heap": https://rustwasm.github.io/wasm-bindgen/contributing/design/js-objects-in-rust.html#anatomy-of-jsvalue.

**Review by relrelb — APPROVED** [review]

**Comment by relrelb** [comment]
Thank you!

**Pull request merged**